### ISSUE TITLE
Sync not working with profiles

### ISSUE THREAD

**Issue by gaetansnl** [GoogleContainerTools/skaffold#2407]
### Expected behavior
I expect sync to work when it's configured in a profile
### Actual behavior
Skaffold trigger full rebuild

### Information

- Skaffold version: 0.33.0
- Operating system: Windows 10

### Steps to reproduce the behavior

1. Sync Working
```
apiVersion: skaffold/v1beta7
kind: Config
build:
  artifacts:
    - image: ********************
      context: .
      sync:
        'packages/server/***': ./
      docker:
        target: server-dev
        dockerfile: ./k8s/image/Dockerfile
  local:
    useBuildkit: true
deploy:
  kustomize:
    path: ./k8s/env-specific/internal-dev
```
2. Sync not working (skaffold trigger full rebuild)
```
apiVersion: skaffold/v1beta7
kind: Config
profiles:
  - name: dev
    build:
      artifacts:
        - image: ***************
          context: .
          sync:
            'packages/server/***': ./
          docker:
            target: server-dev
            dockerfile: ./k8s/image/Dockerfile
      local:
        useBuildkit: true
    deploy:
      kustomize:
        path: ./k8s/env-specific/internal-dev

```

**Comment by dgageot** [comment]
@gaetansnl Sorry, there's an issue with the schema upgrade process.

When your config file is upgraded from `v1beta7` to `v1beta12`, it loses the sync information that are carried by the profiles. If you manually migrate to `v1beta12` and use the new syntax for sync, it should be all good.

ping @corneliusweig [This](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/schema/v1beta9/upgrade.go#L68) should be done on profiles too

**Comment by corneliusweig** [comment]
Sorry, my blunder. Will be fixed by #2415.

Thanks @gaetansnl for reporting!
